src/m365/pp/commands/solution/solution-get.ts

```diff
@@ -0,0 +1,125 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { powerPlatform } from '../../../../utils/powerPlatform';
+import PowerPlatformCommand from '../../../base/PowerPlatformCommand';
+import commands from '../../commands';
+import { Publisher, Solution } from './Solution';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  environment: string;
+  id?: string;
+  name?: string;
+  asAdmin: boolean;
+}
+
+class PpSolutionGetCommand extends PowerPlatformCommand {
+  public get name(): string {
+    return commands.SOLUTION_GET;
+  }
+
+  public get description(): string {
+    return 'Lists a specific solution in a given environment.';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['uniquename', 'version', 'publisher'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environment <environment>'
+      },
+      {
+        option: '-i, --id'
+      },
+      {
+        option: '-n, --name'
+      },
+      {
+        option: '-a, --asAdmin'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      ['id', 'name']
+    );
+  }
+
+  public async commandAction(logger: Logger, args: any): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving a specific solutions for which the user is an admin...`);
+    }
+
+    try {
+      const res: Solution = await this.getSolution(args.options);
+      if (!args.options.output || args.options.output === 'json') {
+        logger.log(res);
+      }
+      else {
+        //converted to text friendly output
+        logger.log({
+          uniquename: res.uniquename,
+          version: res.version,
+          publisher: (res.publisherid as Publisher).friendlyname
+        });
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getSolution(options: Options): Promise<Solution> {
+    const dynamicsApiUrl = await powerPlatform.getDynamicsInstanceApiUrl(options.environment, options.asAdmin);
+    if (options.id) {
+      const requestOptions: AxiosRequestConfig = {
+        url: `${dynamicsApiUrl}/api/data/v9.0/solutions(${options.id})?$expand=publisherid($select=friendlyname)&$select=solutionid,uniquename,version,publisherid,installedon,solutionpackageversion,friendlyname,versionnumber&api-version=9.1`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
```

Comment:
Could we move this request object outside the `if` since we are using it in both `if` and `else`?
In the `if` block we can just modify the object by setting the URL property.